File: ga/gpt3_5_result_2024_6/result/1968B_gpt/44_1968B.c
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut input = String::new();

    // Read number of test cases
    stdin.lock().read_line(&mut input).unwrap();
    let t: i32 = input.trim().parse().unwrap();
    input.clear();

    for _ in 0..t {
        input.clear();

        // Read input values for each test case
        stdin.lock().read_line(&mut input).unwrap();
        let mut values = input.split_whitespace();
        let n: usize = values.next().unwrap().parse().unwrap();
        let m: usize = values.next().unwrap().parse().unwrap();
        
        input.clear();
        stdin.lock().read_line(&mut input).unwrap();
        let sn = input.trim();

        input.clear();
        stdin.lock().read_line(&mut input).unwrap();
        let sm = input.trim();

        let mut i = 0;
        let mut j = 0;

        while i < n && j < m {
            if sn.chars().nth(i) == sm.chars().nth(j) {
                i += 1;
            }
            j += 1;
        }
        
        println!("{}", i);
    }
}

File: ga/gpt3_5_result_2024_6/result/1968B_cpp_gpt/44_1968B.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut input = stdin.lock();
    
    let mut t = String::new();
    input.read_line(&mut t).unwrap();
    let t: i32 = t.trim().parse().unwrap();

    for _ in 0..t {
        let mut line = String::new();
        input.read_line(&mut line).unwrap();
        let mut iter = line.split_whitespace();
        let x: i32 = iter.next().unwrap().parse().unwrap();
        let k: i32 = iter.next().unwrap().parse().unwrap();

        let mut a = String::new();
        input.read_line(&mut a).unwrap();
        let a = a.trim();

        let mut b = String::new();
        input.read_line(&mut b).unwrap();
        let b = b.trim();

        let mut ans = 0;
        for i in b.chars() {
            if ans < a.len() && a.chars().nth(ans).unwrap() == i {
                ans += 1;
            }
        }

        println!("{}", ans);
    }
}